src/main/java/org/mockito/internal/verification/api/VerificationData.java

```diff
@@ -6,26 +6,43 @@
 
 import org.mockito.internal.invocation.InvocationMatcher;
 import org.mockito.invocation.Invocation;
+import org.mockito.invocation.MatchableInvocation;
 
 import java.util.List;
 
 /**
- * @deprecated - This internal class leaks from the public API and therefore was deprecated.
- * Please use {@link org.mockito.verification.VerificationData} instead.
- *
- * Deprecated since 2.2.12
+ * Data needed to perform verification of interactions.
+ * This interface is considered public even though it lives in private package.
+ * In the next major version of Mockito, this class will be moved to public space.
```

Comment:
Actually this still does leak `org.mockito.internal.invocation.InvocationMatcher`